components/TextFields/MDCTextInput.h

```diff
@@ -0,0 +1,159 @@
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@protocol MDCTextInput;
+
+/**
+  Presentation styles for a text input. The style determines specific aspects of the text
+  input, such as sizing, placeholder placement and behavior, ability to show error state,
+  layout, etc.
+
+  NOTE: Showing or hiding the character count / limit label is done via setting characterLimit: when
+  characterLimit == 0, the label is hidden. When characterLimit > 0, the label is shown.
+  */
+typedef NS_ENUM(NSUInteger, MDCTextInputPresentationStyle) {
+  /** Default style with an inline placeholder and character count below text. */
+  MDCTextInputPresentationStyleDefault = 0,
+
+  /**
+   The placeholder text will float above the field when there is content or the field is being
+   edited. The character count is below text.
+   */
+  MDCTextInputPresentationStyleFloatingPlaceholder,
+
+  /** The placeholder is laid out inline and the character count is also inline to the right. */
+  MDCTextInputPresentationStyleFullWidth,
+};
+
+/** Delegate for MDCTextInput size changes. */
+@protocol MDCTextInputLayoutDelegate <NSObject>
+
+@optional
+/**
+  Notifies the delegate that the textInput's content size changed, requiring the size provided.
+
+  @param textInput The text input for which the content size changed.
+  @param size      The size required by the text input to fit all of its content.
+ */
+- (void)textInput:(_Nonnull id<MDCTextInput>)textInput contentSizeChanged:(CGSize)size;
+
+@end
+
+/** Behavior exhibited by Material Design themed text inputs. */
+@protocol MDCTextInput <NSObject>
+
+/** A Boolean value indicating whether the text field is currently in edit mode. */
+@property(nonatomic, readonly, getter=isEditing) BOOL editing;
+
+/** The text to be displayed when no text has been entered. */
+@property(nonatomic, nullable, copy) NSString *placeholder;
+
+/** The text displayed in the text input. */
+@property(nonatomic, nullable, copy) NSString *text;
+
+/**
+ The text displayed under input rectangle, when in error state, that explains the nature of the
+ error.
+  The value of this property controls the state of the text input. When errorText != nil, the text
+  input is in an error state:
+   - The error text appears in a label below the input rectangle's underline with the errorColor as
+ text
+  color.
+   - The input rectangle's underline is colored with the errorColor.
+ */
+@property(nonatomic, nullable, copy, readonly) NSString *errorText;
+
+/**
+ The color used to denote error state in the underline and the errorText's label.
+ Default is red.
+ */
+@property(nonatomic, null_resettable, copy, readonly) UIColor *errorColor;
```

Comment:
Should this be a property on MDCTextView so we can make it accessible via the appearance proxy or ib_inspectable?
Why is it readonly?